youtube_dl/__init__.py

```diff
@@ -141,7 +147,22 @@ def _real_main(argv=None):
             parser.error('auto number start must be positive or 0')
     if opts.usetitle and opts.useid:
         parser.error('using title conflicts with using video ID')
-    if opts.username is not None and opts.password is None:
+    if opts.username is not None and opts.password is None and opts.password_from_keyring is True:
+        # extract domain names, check if all videos are from the same domain.
+        all_domains = set(map(lambda url: compat_urllib_parse.urlparse(url).netloc, all_urls))
+        if len(all_domains) > 1:
+            parser.error('You passed URLs from more than one domain - supplying credentials from command line is not supported in this case.')
+        domain_name = all_domains.pop()
+        password = Secret.password_lookup_sync(LIBSECRET_SCHEMA, {"user-name": opts.username, "domain-name": domain_name}, None)
+        if password is None:
+            print("Password for domain " + domain_name + " and user name " + opts.username + " not found in the keyring.")
+            supplied_password = compat_getpass('Type account password and press [Return]: ')
+            attributes = {"user-name": opts.username, "domain-name": domain_name}
+            label = "Youtube-dl: " + domain_name + " password"
+            Secret.password_store_sync(LIBSECRET_SCHEMA, attributes, Secret.COLLECTION_DEFAULT, label, supplied_password, None)
+        password = Secret.password_lookup_sync(LIBSECRET_SCHEMA, {"user-name": opts.username, "domain-name": domain_name}, None)
+        opts.password = password
```

Comment:
No. You should not shadow the original explicitly provided password.

Reply:
Removed - https://github.com/rg3/youtube-dl/commit/5b295effdd8ff980f319c975a9ec4b69879d5358